src/psyclone/psyir/symbols/intrinsic_symbol.py

```diff
@@ -50,6 +50,13 @@ class IntrinsicSymbol(RoutineSymbol):
     :type kwargs: unwrapped dict.
 
     '''
+    def __init__(self, name, intrinsic, **kwargs):
```

Comment:
Updates docstring.

I agree on simplifying the interface, in fact I think all intrinsic signature information should be moved to IntrinsicSymbol, but when I did this it seemed too large of a change for this already large PR. I added  a TODO to #2541

Reply:
Also note that currently this duplication only happens in tests and inside the create method because we create instrinsics with:
`IntrinsicCall.create(IntrinsicCall.Intrinsic.MAX, [ref_a, ref_b])`

but I agree this should better be:
`Call.create(IntrinsicSymbol("MAX"), [ref_a, ref_b])`
tbd if we need to keep the IntrinsicCall node